### ISSUE TITLE
Ubuntu 23.04 can't install or run veracrypt using 1.25.9 ubuntu 22.04 deb file

### ISSUE THREAD

**Issue by jasonsoooz** [veracrypt/VeraCrypt#1054]
Installing latest veracrypt deb file (1.25.9) via dpkg on Ubuntu 23.04, I run into errors about dependency packages.  I'm not able to install or run veracrypt.  Its mentions unmet dependencies libwxgtk3.0-gtk3-0v5 and libfuse2, that I'm not able to install via apt.  Veracrypt is working on Ubuntu 22.04 and 22.10.  Please help to get working on 23.04.  Thank you.

# Expected behavior

Veracrypt is installed ok and I can run the application.  Like on Ubutnu 22.04 and 22.10

# Observed behavior

1. On cmd line: executing "sudo dpkg -i veracrypt-1.25.9-Ubuntu-22.04-amd64.deb", I get following error
dpkg: dependency problems prevent configuration of veracrypt:
 veracrypt depends on libwxgtk3.0-gtk3-0v5; however:
  Package libwxgtk3.0-gtk3-0v5 is not installed.
 veracrypt depends on libfuse2; however:
  Package libfuse2 is not installed.

2. cmd line: executing "veracrypt" I get following error
veracrypt: error while loading shared libraries: libfuse.so.2: cannot open shared object file: No such file or directory

3. I try to install missing shared library.  cmd line: executing "sudo apt install libfuse2" I get the following error
The following packages have unmet dependencies:
 veracrypt : Depends: libwxgtk3.0-gtk3-0v5 but it is not going to be installed
E: Unmet dependencies. Try 'apt --fix-broken install' with no packages (or specify a solution).

4. I follow apt fix recommendation: cmd line: executing "sudo apt --fix-broken install'
The following packages will be REMOVED:
  veracrypt
0 upgraded, 0 newly installed, 1 to remove and 0 not upgraded.
1 not fully installed or removed.
After this operation, 21.1 MB disk space will be freed.
Do you want to continue? [Y/n] 

5. I select "n" as I don't want to remove veracrypt.

# Steps to reproduce

1. Download latest deb file for Ubuntu: veracrypt-1.25.9-Ubuntu-22.04-amd64.deb
2. On cmd line: execute "sudo dpkg -i veracrypt-1.25.9-Ubuntu-22.04-amd64.deb"
3. Encountered errors: see "Observed behaviour" above
4. I tried with older version of veracrypt and got the same error: veracrypt-1.25.4-Ubuntu-20.04-amd64.deb

# Screenshots

# Your Environment

Please tell us more about your environment

**VeraCrypt version:** 
1.25.9

**Operating system and version:** 
Ubuntu 23.04

**System type:** 
64 bit (amd 64)

**Comment by jasonsoooz** [comment]
I've managed to find a workaround by manually copying the shared libraries (5) from Ubuntu 22.04 to 23.04.  This is what worked for me.

1. Libraries needed
i. libfuse.s0.2.9.9
ii. libwx_gtk3u_adv-3.0.so.0.5.0
iii. libwx_gtk3u_core-3.0.so.0.5.0
iv. libwx_baseu-3.0.so.0.5.0
v. libtiff.so.5.7.0

2. Copy the files from Ubuntu 22.04:/lib/x86_64-linux-gnu to USB

3. Copy the files from USB to Ubuntu 23.04:~/Downloads.  Ensure the file permissions are right (eg -rw-r--r--)

4. Copy the files from Downloads to target location (/lib/x86_64-linux-gnu) and create a symlink for each
```  
$ cd /lib/x86_64-linux-gnu

# 1. copy libfuse file & create symlink
$ sudo cp ~/Downloads/libfuse.s0.2.9.9 .
$ sudo ln -s libfuse.so.2.9.9 libfuse.so.2

# 2. copy libwx_gtk3u_adv file & create symlink
$ sudo cp ~/Downloads/libwx_gtk3u_adv-3.0.so.0.5.0 .
$ sudo ln -s libwx_gtk3u_adv-3.0.so.0.5.0 libwx_gtk3u_adv-3.0.so.0

# 3. copy libwx_gtk3u_core file & create symlink
$ sudo cp ~/Downloads/libwx_gtk3u_core-3.0.so.0.5.0 .
$ sudo ln -s libwx_gtk3u_core-3.0.so.0.5.0 libwx_gtk3u_core-3.0.so.0

# 4. copy libwx_baseu file & create symlink
$ sudo cp ~/Downloads/libwx_baseu-3.0.so.0.5.0 .
$ sudo ln -s libwx_baseu-3.0.so.0.5.0 libwx_baseu-3.0.so.0

# 5. copy libtiff file & create symlink
$ sudo cp ~/Downloads/libtiff.so.5.7.0 .
$ sudo ln -s libtiff.so.5.7.0 libtiff.so.5
```

5. Run veracrypt: $ veracrypt (app runs)

**Comment by jasonsoooz** [comment]
The app works, but it has the downside of not being able to upgrade apt packages with veracrypt installed.  I work around this too.

I get the following error when I run: $ sudo apt upgrade
```
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
You might want to run 'apt --fix-broken install' to correct these.
The following packages have unmet dependencies:
 veracrypt : Depends: libwxgtk3.0-gtk3-0v5 but it is not installable
             Depends: libfuse2 but it is not installed
E: Unmet dependencies. Try 'apt --fix-broken install' with no packages (or specify a solution).
```

When I try to fix apt dependencies, I get
```
$ sudo apt --fix-broken install

Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Correcting dependencies... Done
The following packages will be REMOVED:
  veracrypt
0 upgraded, 0 newly installed, 1 to remove and 0 not upgraded.
1 not fully installed or removed.
After this operation, 21.1 MB disk space will be freed.
Do you want to continue? [Y/n]
```

I select "Y" and it will uninstall veracrypt.  Then apt upgrade works as usual.

Then I install veracrypt again via dpkg when I need it again.  As libraries are already there, it will run fine
$ sudo dpkg -i veracrypt-1.25.9-Ubuntu-22.04-amd64.deb

**Comment by jasonsoooz** [comment]
Please create new deb file for Ubuntu 23.04.  Then we can clean up the manual libraries and not deal with unmet dependencies issue.

Thank you.

**Comment by freebeego** [comment]
Please create new deb file for Ubuntu 23.04

**Comment by archfrog** [comment]
I ran into the same issue, but solved it by installing the latest console-only version of VeraCrypt.  It installs fine and runs just as fine.  I usually use the GUI version, but now I can at least mount and use my VeraCrypt containers :-)

**Comment by jasonsoooz** [comment]
Cool thanks archfrog.  That's a better solution.  I'll close this as the console is good enough.

**Comment by jasonsoooz** [comment]
No, I worked around the GUI issue by using the console.  You're right, GUI issue is still outstanding, so will re-open.

Veracrypt team - please create new deb file for Ubuntu 23.04 GUI.  Thank you

**Comment by Jertzukka** [comment]
The generic installer works for Ubuntu 23.04, though 23.04 doesn't come with libfuse2 out of the box, so you'll need to install it through apt.

**Comment by idrassi** [comment]
I have pushed a commit to fix compatibility issue of .deb with Ubuntu 23.04: https://github.com/veracrypt/VeraCrypt/commit/0fdf2edda77242dbb0fe168197e2aadb402146d1

I have also created a new Ubuntu 23.04 package for VeraCrypt 1.25.9 that uses this change and it is available on 1.25.9 assets on Github and on the official downloads page https://veracrypt.fr/en/Downloads.html
